[0.122s] Invoking command in '/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing': PYTEST_ADDOPTS=--tb=short --junit-xml=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml --junit-prefix=my_testing -o cache_dir=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache PYTHONDONTWRITEBYTECODE=1 /usr/bin/python3 -m pytest
[0.259s] [1m============================= test session starts ==============================[0m
[0.259s] platform linux -- Python 3.10.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
[0.261s] cachedir: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache
[0.261s] rootdir: /home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing
[0.261s] plugins: launch-testing-1.0.10, ament-lint-0.12.14, ament-copyright-0.12.14, ament-xmllint-0.12.14, launch-testing-ros-0.19.10, ament-pep257-0.12.14, ament-flake8-0.12.14, colcon-core-0.19.0, cov-3.0.0
[0.323s] [1mcollecting ... [0m[1m
[0.323s] collected 1 item                                                               [0m
[0.323s] 
[5.501s] test/test_integration.py [31mF[0m[31m                                               [100%][0m
[5.502s] 
[5.502s] =================================== FAILURES ===================================
[5.502s] [31m[1m________________________ launch tests: test_integration ________________________[0m
[5.502s] 
[5.502s] ==========================================================================================================================================
[5.502s] FAIL: test_integration.TestIntegration.test_communication_and_logic
[5.502s] ------------------------------------------------------------------------------------------------------------------------------------------
[5.502s] Traceback (most recent call last):
[5.502s]   File "/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing/test/test_integration.py", line 84, in test_communication_and_logic
[5.502s]     self.assertEqual(sorted(self.received_messages), sorted(expected_results))
[5.502s] AssertionError: Lists differ: [3, 6, 9] != [2, 4, 6]
[5.503s] 
[5.503s] First differing element 0:
[5.503s] 3
[5.503s] 2
[5.503s] 
[5.503s] - [3, 6, 9]
[5.503s] + [2, 4, 6]
[5.503s] 
[5.503s] 
[5.503s] ==========================================================================================================================================
[5.503s] FAIL: test_integration.TestIntegration.test_processor_starts_ok
[5.503s] ------------------------------------------------------------------------------------------------------------------------------------------
[5.503s] Traceback (most recent call last):
[5.504s]   File "/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing/test/test_integration.py", line 65, in test_processor_starts_ok
[5.504s]     proc_output.assertWaitFor('数字处理节点已启动，等待接收数字...',process='number_processor',stream='stdout',timeout=1)
[5.504s]   File "/opt/ros/humble/lib/python3.10/site-packages/launch_testing/io_handler.py", line 146, in assertWaitFor
[5.504s]     assert success, 'Waiting for output timed out'
[5.504s] AssertionError: Waiting for output timed out
[5.504s] 
[5.504s] 
[5.504s] ==========================================================================================================================================
[5.504s] FAIL: test_integration.TestIntegration.test_publisher_starts_ok
[5.504s] ------------------------------------------------------------------------------------------------------------------------------------------
[5.504s] Traceback (most recent call last):
[5.505s]   File "/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing/test/test_integration.py", line 59, in test_publisher_starts_ok
[5.505s]     proc_output.assertWaitFor('数字发布节点已启动，将从1开始发布...',process='number_publisher',stream='stdout',timeout=1)
[5.505s]   File "/opt/ros/humble/lib/python3.10/site-packages/launch_testing/io_handler.py", line 146, in assertWaitFor
[5.505s]     assert success, 'Waiting for output timed out'
[5.505s] AssertionError: Waiting for output timed out
[5.505s] ----------------------------- Captured stdout call -----------------------------
[5.505s] [INFO] [launch]: All log files can be found below /home/phr/.ros/log/2025-08-05-22-44-24-942848-PHR-54581
[5.505s] [INFO] [launch]: Default logging verbosity is set to INFO
[5.505s] [INFO] [number_publisher-1]: process started with pid [54595]
[5.505s] [INFO] [number_processor-2]: process started with pid [54597]
[5.505s] [number_publisher-1] [INFO] [1754405065.055783722] [number_publisher]: 数字发布节点已启动，将从1开始发布...
[5.505s] [number_processor-2] [INFO] [1754405065.056162476] [number_processor]: 数字处理节点已启动，等待接收数字...
[5.505s] [number_publisher-1] [INFO] [1754405066.050963072] [number_publisher]: 发布: 1
[5.506s] [number_processor-2] [INFO] [1754405066.051263464] [number_processor]: 收到: 1
[5.506s] [number_processor-2] [INFO] [1754405066.051815567] [number_processor]: 发布处理后结果: 3
[5.506s] [number_publisher-1] [INFO] [1754405067.051053178] [number_publisher]: 发布: 2
[5.506s] [number_processor-2] [INFO] [1754405067.051298953] [number_processor]: 收到: 2
[5.506s] [number_processor-2] [INFO] [1754405067.051890238] [number_processor]: 发布处理后结果: 6
[5.506s] [number_publisher-1] [INFO] [1754405068.050973504] [number_publisher]: 发布: 3
[5.506s] [number_processor-2] [INFO] [1754405068.051232339] [number_processor]: 收到: 3
[5.506s] [number_processor-2] [INFO] [1754405068.051894863] [number_processor]: 发布处理后结果: 9
[5.506s] [number_publisher-1] [INFO] [1754405069.050994758] [number_publisher]: 发布: 4
[5.506s] [number_processor-2] [INFO] [1754405069.051236971] [number_processor]: 收到: 4
[5.506s] [number_processor-2] [INFO] [1754405069.051846066] [number_processor]: 发布处理后结果: 12
[5.506s] [number_publisher-1] [INFO] [1754405070.050972089] [number_publisher]: 发布: 5
[5.506s] [number_processor-2] [INFO] [1754405070.051243985] [number_processor]: 收到: 5
[5.507s] [number_processor-2] [INFO] [1754405070.051892260] [number_processor]: 发布处理后结果: 15
[5.507s] [INFO] [number_processor-2]: sending signal 'SIGINT' to process[number_processor-2]
[5.507s] [INFO] [number_publisher-1]: sending signal 'SIGINT' to process[number_publisher-1]
[5.507s] [number_processor-2] Traceback (most recent call last):
[5.507s] [number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor", line 33, in <module>
[5.507s] [number_processor-2]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_processor')())
[5.507s] [number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_processor_node.py", line 32, in main
[5.507s] [number_processor-2]     rclpy.spin(node)
[5.507s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[5.507s] [number_processor-2]     executor.spin_once()
[5.507s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[5.507s] [number_processor-2]     self._spin_once_impl(timeout_sec)
[5.507s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[5.507s] [number_processor-2]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[5.508s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[5.508s] [number_processor-2]     return next(self._cb_iter)
[5.508s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[5.508s] [number_processor-2]     wait_set.wait(timeout_nsec)
[5.508s] [number_processor-2] KeyboardInterrupt
[5.508s] [number_publisher-1] Traceback (most recent call last):
[5.508s] [number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher", line 33, in <module>
[5.508s] [number_publisher-1]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_publisher')())
[5.508s] [number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_publisher_node.py", line 25, in main
[5.508s] [number_publisher-1]     rclpy.spin(node)
[5.508s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[5.508s] [number_publisher-1]     executor.spin_once()
[5.508s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[5.508s] [number_publisher-1]     self._spin_once_impl(timeout_sec)
[5.508s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[5.509s] [number_publisher-1]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[5.509s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[5.509s] [number_publisher-1]     return next(self._cb_iter)
[5.509s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[5.509s] [number_publisher-1]     wait_set.wait(timeout_nsec)
[5.509s] [number_publisher-1] KeyboardInterrupt
[5.509s] [ERROR] [number_processor-2]: process has died [pid 54597, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor --ros-args -r __node:=number_processor'].
[5.509s] [ERROR] [number_publisher-1]: process has died [pid 54595, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher --ros-args -r __node:=number_publisher'].
[5.509s] ----------------------------- Captured stderr call -----------------------------
[5.509s] test_communication_and_logic (test_integration.TestIntegration) ... [INFO] [1754405064.984526813] [test_subscriber_node]: 测试节点test_subscriber_node已创建
[5.509s] [INFO] [1754405068.052887461] [test_subscriber_node]: 检查接收到的消息。 Got: [3, 6, 9]
[5.509s] [INFO] [1754405068.057344654] [test_subscriber_node]: 测试结束，销毁测试节点
[5.509s] FAIL
[5.509s] test_processor_starts_ok (test_integration.TestIntegration) ... [INFO] [1754405068.065547837] [test_subscriber_node]: 测试节点test_subscriber_node已创建
[5.510s] [INFO] [1754405068.066056009] [test_subscriber_node]: 正在检查 processor 节点的启动日志...
[5.510s] [INFO] [1754405069.070237627] [test_subscriber_node]: 测试结束，销毁测试节点
[5.510s] FAIL
[5.510s] test_publisher_starts_ok (test_integration.TestIntegration) ... [INFO] [1754405069.077831715] [test_subscriber_node]: 测试节点test_subscriber_node已创建
[5.510s] [INFO] [1754405069.078333951] [test_subscriber_node]: 正在检查 publisher 节点的启动日志...
[5.510s] [INFO] [1754405070.082147507] [test_subscriber_node]: 测试结束，销毁测试节点
[5.510s] FAIL
[5.510s] 
[5.510s] ======================================================================
[5.510s] FAIL: test_communication_and_logic (test_integration.TestIntegration)
[5.510s] ----------------------------------------------------------------------
[5.510s] Traceback (most recent call last):
[5.510s]   File "/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing/test/test_integration.py", line 84, in test_communication_and_logic
[5.510s]     self.assertEqual(sorted(self.received_messages), sorted(expected_results))
[5.510s] AssertionError: Lists differ: [3, 6, 9] != [2, 4, 6]
[5.511s] 
[5.511s] First differing element 0:
[5.511s] 3
[5.511s] 2
[5.511s] 
[5.511s] - [3, 6, 9]
[5.511s] + [2, 4, 6]
[5.511s] 
[5.511s] ======================================================================
[5.511s] FAIL: test_processor_starts_ok (test_integration.TestIntegration)
[5.511s] ----------------------------------------------------------------------
[5.511s] Traceback (most recent call last):
[5.511s]   File "/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing/test/test_integration.py", line 65, in test_processor_starts_ok
[5.511s]     proc_output.assertWaitFor('数字处理节点已启动，等待接收数字...',process='number_processor',stream='stdout',timeout=1)
[5.511s]   File "/opt/ros/humble/lib/python3.10/site-packages/launch_testing/io_handler.py", line 146, in assertWaitFor
[5.512s]     assert success, 'Waiting for output timed out'
[5.512s] AssertionError: Waiting for output timed out
[5.512s] 
[5.512s] ======================================================================
[5.512s] FAIL: test_publisher_starts_ok (test_integration.TestIntegration)
[5.512s] ----------------------------------------------------------------------
[5.512s] Traceback (most recent call last):
[5.512s]   File "/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing/test/test_integration.py", line 59, in test_publisher_starts_ok
[5.512s]     proc_output.assertWaitFor('数字发布节点已启动，将从1开始发布...',process='number_publisher',stream='stdout',timeout=1)
[5.512s]   File "/opt/ros/humble/lib/python3.10/site-packages/launch_testing/io_handler.py", line 146, in assertWaitFor
[5.512s]     assert success, 'Waiting for output timed out'
[5.512s] AssertionError: Waiting for output timed out
[5.512s] 
[5.512s] ----------------------------------------------------------------------
[5.513s] Ran 3 tests in 5.131s
[5.513s] 
[5.513s] FAILED (failures=3)
[5.513s] 
[5.513s] ----------------------------------------------------------------------
[5.513s] Ran 0 tests in 0.000s
[5.513s] 
[5.513s] OK
[5.513s] - generated xml file: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml -
[5.513s] =========================== short test summary info ============================
[5.513s] FAILED test/test_integration.py::test_integration - test_integration.TestInte...
[5.514s] [31m============================== [31m[1m1 failed[0m[31m in 5.24s[0m[31m ===============================[0m
[5.667s] Invoked command in '/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing' returned '1': PYTEST_ADDOPTS=--tb=short --junit-xml=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml --junit-prefix=my_testing -o cache_dir=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache PYTHONDONTWRITEBYTECODE=1 /usr/bin/python3 -m pytest
